scss/components/_badge.scss

```diff
@@ -18,6 +18,10 @@ $badge-color: $white !default;
 /// @type Color
 $badge-color-alt: $black !default;
 
+/// Coloring classes. The names in this list output classes in your CSS, like `.secondary`, `.success`, and so on. Each value must also be in the `$foundation-palette` map.
+/// @type List
+$badge-classes: ( secondary success warning alert ) !default;
```

Comment:
> But if components have their own color/class maps, there's no reason for $foundation-palette to be a map.

To iterate on it too, like we did with `$foundation-palette`, but with their own colors.